packages/core/src/state-management/anchor-resuming-service.ts

```diff
@@ -0,0 +1,64 @@
+import { StreamID } from '@ceramicnetwork/streamid'
+import { AnchorRequestStoreListResult } from '../store/anchor-request-store.js'
+import { AnchorStatus, DiagnosticsLogger, LogStyle, SyncOptions } from '@ceramicnetwork/common'
+import { Repository } from './repository.js'
+import PQueue from 'p-queue'
+
+const RESUME_QUEUE_CONCURRENCY = 30
+const RESUME_BATCH_SIZE = RESUME_QUEUE_CONCURRENCY * 5
+
+export class AnchorResumingService {
+  /**
+   * Resume running states from anchor request store
+   */
+  readonly resumeQ: PQueue
+
+  /**
+   * true iff the repository is in the process of closing
+   *
+   * @private
+   */
+  #shouldBeClosed = false
+
+  constructor(private readonly logger: DiagnosticsLogger) {
+    this.resumeQ = new PQueue({ concurrency: RESUME_QUEUE_CONCURRENCY })
+  }
+
+  async resumeRunningStatesFromAnchorRequestStore(repository: Repository): Promise<void> {
+    this.#shouldBeClosed = false
+    const currentTimestamp = Date.now()
+    let gt: StreamID | undefined = undefined
+    let batch = new Array<AnchorRequestStoreListResult>()
+    do {
+      batch = await repository.anchorRequestStore.list(RESUME_BATCH_SIZE, gt)
+      batch.map((listResult) => {
+        this.resumeQ.add(async () => {
+          const stream$ = await repository.load(listResult.key, {
+            sync: SyncOptions.PREFER_CACHE,
+          })
+          if ([AnchorStatus.FAILED, AnchorStatus.ANCHORED].includes(stream$.value.anchorStatus)) {
+            await repository.anchorRequestStore.remove(stream$.id)
+            return
+          }
+          // TODO: CDB-2010 Do we want to do the actual anchor request here? Or do we just want to start polling? If the latter, than .load(..) doesn't seem to start polling as of now.
+          await repository.stateManager.anchor(stream$)
+          if (listResult.value.timestamp > currentTimestamp) {
```

Comment:
can you help me understand what this check is doing and why it's necessary?

Reply:
When we were discussing this project during the spec review, you said that it was important for you to check, if the node's time is moving backwards. This is what this check is supposed to verify.

You think I should remove it or change it, @stbrody ?

Reply:
ah I see.  I guess I had imagined that being tracked and handled separately.  Tying the system clock check in with the data in the state store worries me for the reasons I mentioned in my other comment.  Can we have a separate ticket for the check around the time moving backward and do some brainstorming on the best and safest approach to address that?  That way we don't hold up this PR on trying to figure that all out